pkg/chart/chart_test.go

```diff
@@ -409,9 +409,13 @@ type fakeHTTPClient struct {
 	chartURLs []string
 	index     *repo.IndexFile
 	userAgent string
+	// TODO(absoludity): perhaps switch to use httptest instead of our own fake?
+	requests []*http.Request
 }
 
 func (f *fakeHTTPClient) Do(h *http.Request) (*http.Response, error) {
+	// Record the request for later test assertions.
+	f.requests = append(f.requests, h)
 	if f.userAgent != "" && h.Header.Get("User-Agent") != f.userAgent {
 		return nil, fmt.Errorf("Wrong user agent: %s", h.Header.Get("User-Agent"))
```

Comment:
Since we're recording the requests now - and can assert on the actual requests in the context of the test, we could simplify this fake `Do` implementation, removing the test assertions like this one? That is, rather than the test not actually doing any assertion (other than that an error was not returned), we can assert the user-agent was set as expected in the context of the test, rather than in this fake code (which has no access to the `t.Testing` to call `t.Errorf` etc). See what you think - I won't do it in this PR, but just a thought.

Reply:
that sounds good to me, the only reason for doing the assertion here is that we couldn't do it in the test. If we can I am happy to do it.